packages/gestalt/src/Sheet.js

```diff
@@ -0,0 +1,283 @@
+      </Heading>
+    </Box>
+  );
+}
+
+/*
+Currently supported sheet animations:  
+- g-sheet-slide-in-ltr: slides the sheet in, from the right edge. Screen direction: ltr
+- g-sheet-slide-in-rtl: slides the sheet in, from the left edge. Screen direction: rtl
+- g-sheet-slide-out-ltr: slides the sheet out, from the right edge. Screen direction: ltr
+- g-sheet-slide-out-rtl: slides the sheet out, from the left edge. Screen direction: rtl
+*/
+const getAnimation = (moment: 'in' | 'out') => {
+  const documentDirection = document?.documentElement?.dir || 'ltr';
+  const animations = {
+    in: {
+      backdrop: 'g-backdrop-fade-in',
+      sheet: `g-sheet-slide-in-${documentDirection}`,
+    },
+    out: {
+      backdrop: 'g-backdrop-fade-out',
+      sheet: `g-sheet-slide-out-${documentDirection}`,
+    },
+  };
+  return animations[moment];
+};
+
+const SheetWithForwardRef: React$AbstractComponent<
+  Props,
+  HTMLDivElement
+> = forwardRef<Props, HTMLDivElement>(function Sheet(props, sheetRef): Node {
+  const {
+    accessibilityDismissButtonLabel,
+    accessibilitySheetLabel,
+    children,
+    closeOnOutsideClick = true,
+    onDismiss,
+    footer,
+    heading,
+    size = 'sm',
+  } = props;
+
+  const [showTopShadow, setShowTopShadow] = useState(false);
+  const [showBottomShadow, setShowBottomShadow] = useState(false);
+  const [currentAnimation, setCurrentAnimation] = useState(getAnimation('in'));
+  const containerRef = useRef<?HTMLDivElement>(null);
+  const contentRef = useRef<?HTMLDivElement>(null);
+  const dismissButtonRef = useRef<?HTMLButtonElement>(null);
+  const shouldReduceMotion = useReducedMotion();
+
+  const startDismiss = useCallback(() => {
+    if (shouldReduceMotion) {
+      onDismiss();
+    } else {
+      setCurrentAnimation(getAnimation('out'));
+      if (containerRef.current) {
+        containerRef.current.addEventListener('animationend', onDismiss);
+      }
+    }
+  }, [onDismiss, shouldReduceMotion]);
+
+  useEffect(() => {
+    function handleKeyUp(event: {| keyCode: number |}) {
+      if (event.keyCode === ESCAPE_KEY_CODE) {
+        startDismiss();
+      }
+    }
+
+    window.addEventListener('keyup', handleKeyUp);
+    return function cleanup() {
+      window.removeEventListener('keyup', handleKeyUp);
+    };
+  }, [startDismiss]);
+
+  const updateShadows = () => {
+    const target = contentRef.current;
+    if (!target) {
+      return;
+    }
+    const hasVerticalScrollbar = target.clientHeight < target.scrollHeight;
+    setShowTopShadow(hasVerticalScrollbar && target.scrollTop > 0);
+    setShowBottomShadow(
+      hasVerticalScrollbar &&
+        target.offsetHeight + target.scrollTop < target.scrollHeight
+    );
+  };
+
+  useEffect(() => {
+    updateShadows();
+    window.addEventListener('resize', updateShadows);
+    return () => {
+      window.removeEventListener('resize', updateShadows);
+    };
+  }, []);
+
+  useEffect(() => {
+    if (dismissButtonRef.current) {
```

Comment:
I see. I think probably the easiest solution would to remove this and then it will just focus on the Sheet wrapper (not displaying an outline). I will explore using `useFocusVisible` so it gets an outline from the keyboard navigation but not from the mouse.